torch/lib/THCS/generic/THCSTensorMath.cu

```diff
@@ -460,6 +462,104 @@ void THCSTensor_(cmul)(THCState *state, THCSTensor *r_, THCSTensor *t_, THCSTens
   THCSTensor_(free)(state, src);
 }
 
+
+ void THCTensor_(spbqrfactsolve)(THCState *state, THCTensor *rx_, THCTensor *b, THCIndexTensor *Ai, THCTensor *Av, THLongStorage *Asz) {
+#if defined(THCS_REAL_IS_FLOAT) || defined(THCS_REAL_IS_DOUBLE)
+   THCAssertSameGPU(THCSTensor_(checkGPU)(state, 1, 3, rx_, b, Av));
+
+   THArgCheck(b->nDimension == 2, 2,
+              "Batch tensor b expected, got %dD tensor", b->nDimension);
+   THArgCheck(Av->nDimension == 2, 2,
+              "Batch tensor Av expected, got %dD tensor", Av->nDimension);
+
+   long nBatch = THCTensor_(size)(state, Av, 0);
+   long nnz = THCTensor_(size)(state, Av, 1);
+   long m = Asz->data[0];
+   long n = Asz->data[1];
+
+   THCTensor_(resize2d)(state, rx_, nBatch, m);
+
+   THCIndexTensor *rowIndices = THCIndexTensor_(newSelect)(state, Ai, 0, 0);
+   THCIndexTensor *colIndices = THCIndexTensor_(newSelect)(state, Ai, 0, 1);
+   THCudaIntTensor *csr = THCSTensor_(toCSR)(state, rowIndices, m, nnz);
+   THCudaIntTensor *colIndicesInt = THCudaIntTensor_newWithSize1d(state, colIndices->size[0]);
+   THCudaIntTensor_copyCudaLong(state, colIndicesInt, colIndices);
+
+   cusparseStatus_t cusparse_status;
+   cusolverStatus_t cusolver_status;
+   cudaError_t cuda_status;
+   csrqrInfo_t info;
+
+   cusolverSpHandle_t cusolverH;
+   cusolver_status = cusolverSpCreate(&cusolverH);
+   THAssert(CUSOLVER_STATUS_SUCCESS == cusolver_status);
+
+   cusparseMatDescr_t descrA;
+   size_t size_qr = 0;
+   size_t size_internal = 0;
+   void *buffer_qr = NULL;
+
+   cusparse_status = cusparseCreateMatDescr(&descrA);
+   THAssert(cusparse_status == CUSPARSE_STATUS_SUCCESS);
+
+   cusparseSetMatType(descrA, CUSPARSE_MATRIX_TYPE_GENERAL);
+   cusparseSetMatIndexBase(descrA, CUSPARSE_INDEX_BASE_ZERO);
```

Comment:
I added a conditional around this line (but am not sure if it's working with 1-based indexing)